.eslintrc.json

```diff
@@ -7,21 +7,40 @@
   },
   "extends": [
     "plugin:react/recommended",
-    "standard"
+    "standard",
+    "plugin:import/typescript"
   ],
   "ignorePatterns": "scripts",
-  "overrides": [{
-    "files": ["*.ts", "*.tsx"],
-    "rules": {
-      "semi": "off",
-      "Extra semicolon": "off",
-      "import/prefer-default-export": "off",
-      // temporary fix from 'typescript-elint' docs
-      // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
-      "no-unused-vars": "off",
-      "@typescript-eslint/no-unused-vars": ["error"]
+  "overrides": [
+    {
+      "files": ["**/*.ts", "**/*.tsx"],
+      "env": { "browser": true, "es6": true, "node": true },
+      "extends": [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended"
+      ],
+      "globals": { "Atomics": "readonly", "SharedArrayBuffer": "readonly" },
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "ecmaFeatures": { "jsx": true },
+        "ecmaVersion": 2018,
+        "sourceType": "module",
+        "project": "./tsconfig.eslint.json"
+      },
+      "plugins": ["@typescript-eslint"],
+      "rules": {
+        "import/prefer-default-export": 0,
+        // temporary fix from 'typescript-elint' docs  https://bit.ly/2XhUE8h
+        "no-unused-vars": 0,
+        "@typescript-eslint/no-unused-vars": [2],
+        "@typescript-eslint/no-explicit-any": 0,
+        "@typescript-eslint/explicit-module-boundary-types": 0,
+        "@typescript-eslint/ban-ts-comment": 1
```

Comment:
Do these only work as `overrides` rather than normal `rules`?

Reply:
They do work as overrides.

Reply:
I wasn’t asking if they work as overrides. I was asking if they *only* work as overrides. If so, why? If not, why? Please provide a more thorough response.

Reply:
The rules under the overrides section are supposed to work `only` for the `matched files`. You can refer [this](https://eslint.org/docs/user-guide/configuring#disabling-rules-only-for-a-group-of-files
) more details.

Reply:
Yes, I understand what `overrides` does. I am asking why it is needed here. You still have explained the reason. What happens when they are added as regular rules?